lib/ansible/modules/windows/win_scheduled_task.ps1

```diff
@@ -1,254 +1,1146 @@
+            }
 
-        if (-not $check_mode) {
-            Register-ScheduledTask -Action $action -Trigger $trigger -TaskName $name -Description $description -TaskPath $path -Settings $settings @registerRunOptionParams
         }
+        if ($week_value -eq 0) {
+            $week_value = $null
+        }
+        $trigger.weeks_of_month = $week_value
+    }
+    if ($trigger.ContainsKey("months_of_year")) {
+        $months = $trigger.months_of_year
+        if ($months -is [String]) {
+            $months = $months.Split(",").Trim()
+        } elseif ($months -isnot [Array]) {
+            $months = @($months)
+        }
+
+        $month_value = 0
+        foreach ($month in $months) {
+            # https://msdn.microsoft.com/en-us/library/windows/desktop/aa382064(v=vs.85).aspx
+            switch ($month) {
+                january { $month_value = $month_value -bor 0x01 }
+                february { $month_value = $month_value -bor 0x02 }
+                march { $month_value = $month_value -bor 0x04 }
+                april { $month_value = $month_value -bor 0x08 }
+                may { $month_value = $month_value -bor 0x10 }
+                june { $month_value = $month_value -bor 0x20 }
+                july { $month_value = $month_value -bor 0x40 }
+                august { $month_value = $month_value -bor 0x80 }
+                september { $month_value = $month_value -bor 0x100 }
+                october { $month_value = $month_value -bor 0x200 }
+                november { $month_value = $month_value -bor 0x400 }
+                december { $month_value = $month_value -bor 0x800 }
+                default { Fail-Json -obj $result -message "invalid month name '$month', please specify full month name" }
+            }
+        }
+        if ($month_value -eq 0) {
+            $month_value = $null
+        }
+        $trigger.months_of_year = $month_value
+    }
+    $triggers[$i] = $trigger
+}
+
+# add \ to start of path if it is not already there
+if (-not $path.StartsWith("\")) {
+    $path = "\$path"
+}
+# ensure path does not end with \ if more than 1 char
+if ($path.EndsWith("\") -and $path.Length -ne 1) {
+    $path = $path.Substring(0, $path.Length - 1)
+}
+
+########################
+### START CODE BLOCK ###
+########################
+$service = New-Object -ComObject Schedule.Service
+try {
+    $service.Connect()
+} catch {
+    Fail-Json -obj $result -message "failed to connect to the task scheduler service: $($_.Exception.Message)"
+}
+
+# check that the path for the task set exists, create if need be
+try {
+    $task_folder = $service.GetFolder($path)
+} catch {
+    $task_folder = $null
+}
 
+# try and get the task at the path
+$task = Test-TaskExists -task_folder $task_folder -name $name
+
+$task_path = $null
+if ($path.EndsWith("`\")) {
+    $task_path = "$path$name"
+} else {
+    $task_path = "$path\$name"
+}
```

Comment:
Isn't there something similar to `os.path.join()` that would abstract this away in PS ?

Reply:
Looks like `Join-Path` does the trick, I'll update the PR.

Thanks for the suggestion.